module/os/linux/zfs/abd.c

```diff
@@ -802,15 +978,99 @@ abd_alloc_for_io(size_t size, boolean_t is_metadata)
 	return (abd_alloc(size, is_metadata));
 }
 
+/*
+ * Create an ABD that will be the head of a list of ABD's. This is used
+ * to "chain" scatter/gather lists together when constructing aggregated
+ * IO's. To free this abd, abd_put() must be called.
```

Comment:
I am perfectly happy changing this to be abd_free(). It would be no problem at all to change this and it makes more sense with the naming scheme of alloc.

Reply:
I did wind up changing this. abd_free() should be called with the returned ABD after abd_alloc_gang_abd() is called.